### ISSUE TITLE
Fails to copy hosts file on Android L Dev Preview

### ISSUE THREAD

**Issue by jwoglom** [Free-Software-for-Android/AdAway#526]
Adaway fails to copy the hosts file when running on the new Android L developer preview. Here is the exception shown in logcat on my Nexus 5:

<code>E/RootCommands( 4713): Exception
E/RootCommands( 4713): org.sufficientlysecure.rootcommands.util.RootAccessDeniedException: stdout line is null! Access was denied or this executeable is not a shell!
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Shell.<init>(Shell.java:147)
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Shell.startRootShell(Shell.java:62)
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Shell.startRootShell(Shell.java:74)
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Remounter.getMounts(Remounter.java:166)
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Remounter.remount(Remounter.java:66)
E/RootCommands( 4713):  at org.sufficientlysecure.rootcommands.Toolbox.remount(Toolbox.java:733)
E/RootCommands( 4713):  at org.adaway.util.ApplyUtils.copyHostsFile(ApplyUtils.java:195)
E/RootCommands( 4713):  at org.adaway.service.ApplyService.apply(ApplyService.java:399)
E/RootCommands( 4713):  at org.adaway.service.ApplyService.doWakefulWork(ApplyService.java:100)
E/RootCommands( 4713):  at com.commonsware.cwac.wakeful.WakefulIntentService.onHandleIntent(WakefulIntentService.java:100)
E/RootCommands( 4713):  at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/RootCommands( 4713):  at android.os.Handler.dispatchMessage(Handler.java:102)
E/RootCommands( 4713):  at android.os.Looper.loop(Looper.java:135)
E/RootCommands( 4713):  at android.os.HandlerThread.run(HandlerThread.java:61)
E/AdAway  ( 4713): Remounting failed in finally! Probably not a problem!
E/AdAway  ( 4713): Exception:
E/AdAway  ( 4713): org.adaway.util.CommandException
E/AdAway  ( 4713):      at org.adaway.util.ApplyUtils.copyHostsFile(ApplyUtils.java:190)
E/AdAway  ( 4713):      at org.adaway.service.ApplyService.apply(ApplyService.java:399)
E/AdAway  ( 4713):      at org.adaway.service.ApplyService.doWakefulWork(ApplyService.java:100)
E/AdAway  ( 4713):      at com.commonsware.cwac.wakeful.WakefulIntentService.onHandleIntent(WakefulIntentService.java:100)
E/AdAway  ( 4713):      at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/AdAway  ( 4713):      at android.os.Handler.dispatchMessage(Handler.java:102)
E/AdAway  ( 4713):      at android.os.Looper.loop(Looper.java:135)
E/AdAway  ( 4713):      at android.os.HandlerThread.run(HandlerThread.java:61)
I/BootReceiver(  800): Copying /data/tombstones/tombstone_07 to DropBox (SYSTEM_TOMBSTONE)
I/notification_cancel(  800): [10093,4713,org.adaway,20,NULL,0,0,64,8,NULL]
I/ActivityManager(  800): START u0 {flg=0x10000004 cmp=org.adaway/.ui.BaseActivity (has extras)} from pid 4713 on display 0
I/am_new_intent(  800): [0,36576106,2,org.adaway/.ui.BaseActivity,NULL,NULL,NULL,268435460]
I/am_destroy_service(  800): [0,809020872,4713]
E/libsigchain( 8367): Unable to find next sigaction in signal chain
F/libc    ( 8367): Fatal signal 6 (SIGABRT), code -6 in tid 8367 (main)
</code>

This seems to be a common problem among root apps on the dev preview -- I have heard that there are additional permission or verification checks that need to be done before commands can be run as root. As someone who loves using AdAway, it would be great if you could fix this issue! Better to fix it now before the final version starts to roll out onto devices.

**Comment by nyuszika7h** [comment]
This issue seems no longer relevant, as Lollipop has been released for a while, and AdAway 2.9.2 works fine on my Nexus 5.
